openpype/modules/deadline/plugins/publish/submit_publish_job.py

```diff
@@ -457,9 +457,15 @@ def _create_instances_for_aov(self, instance_data, exp_files):
 
             cam = [c for c in cameras if c in col.head]
             if cam:
-                subset_name = '{}_{}_{}'.format(group_name, cam, aov)
+                if aov:
+                    subset_name = '{}_{}_{}'.format(group_name, cam, aov)
+                else:
+                    subset_name = '{}_{}'.format(group_name, cam)
             else:
-                subset_name = '{}_{}'.format(group_name, aov)
+                if aov:
+                    subset_name = '{}_{}'.format(group_name, aov)
+                else:
+                    subset_name = '{}'.format(group_name)
```

Comment:
Wouldn't this have been more succinct with something like this?
```python
    subset_name = group_name
    if cam:
        subset_name += "_{}".format(cam)
    if aov:
        subset_name += "_{}".format(aov)
```

Or even:
```python
    subset_name = group_name
    for suffix_element in [cam, aov]:
        if suffix_element:
            subset_name += "_{}".format(suffix_element)
```

Or even:
```python
    parts = [part for part in [group_name, cam, aov] if part]
    subset_name = "_".join(parts)
```

Ok, that last one might be a bit too far for readability 😆